keps/sig-auth/3299-kms-v2-improvements/README.md

```diff
@@ -0,0 +1,518 @@
+Pick one of these and delete the rest.
+-->
+
+- Feature gate
+  - Feature gate name: `KMSv2`
+  - Components depending on the feature gate:
+    - kube-apiserver
+
+```go
+FeatureSpec{
+	Default: false,
+	LockToDefault: false,
+	PreRelease: featuregate.Alpha,
+}
+```
+
+###### Does enabling the feature change any default behavior?
+
+No.  The v2 API is new in the v1.25 release.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, via the `KMSv2` feature gate. Disabling this gate without first doing a storage migration to use a different encryption at rest mechanism will result in data loss.
+
+### Monitoring Requirements
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [x] Other (treat as last resort)
+  - Details: Logs in kube-apiserver, kms-plugin and KMS will be logged with the corresponding `observedKeyID`, `metadata`, and `UID`.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+There should be no impact on the SLO with this change.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [x] Other (treat as last resort)
+  - Details: Logs in kube-apiserver, kms-plugin and KMS will be logged with the corresponding `observedKeyID`, `metadata`, and `UID`.
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+Yes, the new KMS v2 gRPC API.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+Yes, the new KMS v2 gRPC types.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No, the v2 API is new.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No.
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+- ETCD data encryption with external kms-plugin is unavailable
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Alternatives
```

Comment:
We need this optional section to note the need for a new repo for the reference implementation, probably synced out from the k/k staging dir.

https://github.com/kubernetes/enhancements/blob/06a35f811450b98fe85812c650fe16f70c03d8f8/keps/NNNN-kep-template/README.md?plain=1#L782-L788

Reply:
added